src/common/hooks/useShouldZeroApprove/shouldZeroApprove.ts

```diff
@@ -1,44 +1,71 @@
 import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
 import { Erc20 } from 'legacy/abis/types'
 import { ApprovalState } from 'legacy/hooks/useApproveCallback'
+import { Nullish } from 'types'
 
-interface ShouldZeroApproveParams {
+interface ShouldZeroApproveBaseParams {
+  tokenContract: Nullish<Erc20>
+  spender: Nullish<string>
+  amountToApprove: Nullish<CurrencyAmount<Currency>>
+}
+
+interface ShouldZeroApproveBundleParams extends ShouldZeroApproveBaseParams {
+  isBundle: true
+  approvalState?: undefined
+}
+
+interface ShouldZeroApproveNonBundleParams extends ShouldZeroApproveBaseParams {
   approvalState: ApprovalState
-  tokenContract: Erc20 | null
-  spender: string | undefined
-  amountToApprove: CurrencyAmount<Currency> | undefined
+  isBundle?: undefined
 }
 
+type ShouldZeroApproveParams = ShouldZeroApproveBundleParams | ShouldZeroApproveNonBundleParams
+
 export async function shouldZeroApprove({
   approvalState,
   tokenContract,
   spender,
   amountToApprove,
+  isBundle,
 }: ShouldZeroApproveParams) {
-  const shouldApprove = approvalState === ApprovalState.NOT_APPROVED || approvalState === ApprovalState.PENDING
-  if (tokenContract && spender && amountToApprove) {
-    if (!shouldApprove) {
-      return false
-    }
+  if (!tokenContract) {
+    return false
+  }
+
+  if (!spender) {
+    return false
+  }
 
+  if (!amountToApprove) {
+    return false
+  }
+
+  const shouldApprove =
+    isBundle || approvalState === ApprovalState.NOT_APPROVED || approvalState === ApprovalState.PENDING
+
+  if (!shouldApprove) {
+    return false
+  }
```

Comment:
Just a nitpick, no need to change it if you don't want to.
Why split up the conditions in individual ifs?
I would personally do something like:

``` ts
if (!tokenContract || !spender || !amountToApprove || !shouldApprove) {
  return false
}
```